### PR TITLE
Add rebalance listener

### PR DESCRIPTION
**Pull request by MSDehghan**
Add ability to set re-balance listener for consumer.

### CODE REVIEW

File: src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java [hunk 1]
```diff
@@ -132,13 +131,19 @@
...
 
     /**
      * List of partitions assigned to consumer after first connection to Kafka or rebalancing. Assigned partitions
      * cached and will be used later.
      */
-    private List<TopicPartition> assignedPartitions;
+    private final List<TopicPartition> assignedPartitions;
+
+    /**
+     * The callback which is called when rebalance happens.
```
Comment (bozorgzadeh) [inline src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java]: 1. Replace:
The callback ...  ->  The callback provided by the client ...

2. Move it up below the internalRebalanceListener

File: src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java [hunk 2]
```diff
@@ -132,13 +131,19 @@
      * We have used a same callback that does nothing other than writing simple logs and resetting
      * the {@link #offsetTracker}.
```
Comment (bozorgzadeh) [inline src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java]: Replace: 
We have used ... -> It resets the {@link #offsetTracker} and also calls the rebalance listener provided by the client.

Comment (MSDehghan) [reply src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java]: Done

File: src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java (hunk 1 above)
Comment (MSDehghan) [reply src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java]: Done

**Pull request closed without merging**